### DISCUSSION TITLE
Upserting records to s3: .s3.merge_upsert_table vs overwrite partition parameter in s3 write functions

### DISCUSSION THREAD

**Discussion by AdrianoNicolucci** [aws/aws-sdk-pandas#635] [Q&A] [answered]
I would like to upsert records in a parquet table without overwriting the entire partition. 

Would the s3.merge_upsert_table function achieve this or is the s3.merge_upsert_table method just the same as using s3.to_parquet method with the parameter  _overwrite_partitions_ (Partition Upsert) enabled?

**Comment by jaidisido** [comment, marked as answer]
Thank you for discussing this. The current `s3.merge_upsert_table` method is very naive and lacks numerous features. As you have pointed out in your raised issues, loading the entire table into a data frame is inefficient and the method does not support partitioning.

I have already covered this in this [issue](https://github.com/awslabs/aws-data-wrangler/issues/566#issuecomment-791278375) but to summarise once more:
- AWS Glue [Governed Tables](https://aws.amazon.com/blogs/big-data/part-1-effective-data-lakes-using-aws-lake-formation-part-1-getting-started-with-governed-tables/?nc1=b_rp) is a new feature (currently in preview) which allows multiple users to concurrently insert, delete, and modify rows across tables within ACID transactions
- We are currently working on a [PR](https://github.com/awslabs/aws-data-wrangler/pull/571) to integrate it as soon as it's available
- You can find examples of what it will allow you to do [here](https://github.com/awslabs/aws-data-wrangler/blob/main-governed-tables/tutorials/029%20-%20Lake%20Formation%20Governed%20Tables.ipynb). But it should cover all of the elements that you have raised (inefficiency, partitioning...)
- Once released, the `s3.merge_upsert_table` method will be deprecated in Wrangler

**Reply by AdrianoNicolucci** [reply]
great thanks for the reply! Good to know that merge_upsert_table is being deprecated. I'll wait for governed tables to be released.
